app/scripts/controllers/preferences.js

```diff
@@ -302,14 +302,19 @@ class PreferencesController {
     this.store.updateState({ identities, accountTokens })
   }
 
-  /*
+  /**
    * Synchronizes identity entries with known accounts.
    * Removes any unknown identities, and returns the resulting selected address.
    *
    * @param {Array<string>} addresses - known to the vault.
    * @returns {Promise<string>} - selectedAddress the selected address.
    */
   syncAddresses (addresses) {
+
+    if (!Array.isArray(addresses) || addresses.length === 0) {
+      return this.getSelectedAddress()
+    }
+
```

Comment:
There is logic checking for this in `_onKeyringControllerUpdate`, too, but I think we should make sure it never happens by also doing it here.

Reply:
It might be better to throw an error in this case rather than ignore the request. This seems like a fail-loud type of scenario, rather than silently ignore and continue.

Reply:
Good call, fixed